x-pack/plugins/security/public/management/roles/edit_role/privileges/es/index_privilege_form.tsx

```diff
@@ -329,14 +329,29 @@ export class IndexPrivilegeForm extends Component<Props, State> {
                 />
               }
               fullWidth={true}
+              data-test-subj={`queryInput${this.props.formIndex}`}
             >
-              <EuiTextArea
-                data-test-subj={`queryInput${this.props.formIndex}`}
-                style={{ resize: 'none' }}
-                fullWidth={true}
-                value={indexPrivilege.query}
+              <CodeEditor
```

Comment:
Can we use the `CodeEditorField` component instead? It renders the code editor in the style of a text field so looks a bit more consistent with the rest of the form. We probably have to update it to accept the `fullWidth` prop and pass that through to fill the page but otherwise it should work.

One UX issue I noticed is that the page stops scrolling when the cursor is over the code editor. An element should scroll the page if it doesn't have any scrollbars and it's quite noticeable with the element growing so tall now.

Reply:
Yeah let me give this another try. The `CodeEditorField` component wasn't respecting the `fullWidth` settings of this screen, even when passing the prop down "correctly". The result was a misaligned code editor that looked really bad

Reply:
update: `CodeEditorField` works for me! Not sure what my problem was the first time around...thanks for calling this out